packages/plugin-defender-hardhat/src/index.ts

```diff
@@ -29,13 +29,16 @@ export interface HardhatDefenderConfig {
 }
 
 extendConfig((config: HardhatConfig, userConfig: Readonly<HardhatUserConfig>) => {
-  if (!userConfig.defender || !userConfig.defender.apiKey || !userConfig.defender.apiSecret) {
-    const sampleConfig = JSON.stringify({ apiKey: 'YOUR_API_KEY', apiSecret: 'YOUR_API_SECRET' }, null, 2);
-    console.warn(
-      `Defender API key and secret are not set. Add the following to your hardhat.config.js exported configuration:\n\n${sampleConfig}\n`,
-    );
-  }
-  config.defender = userConfig.defender;
+  config.defender = lazyObject((): HardhatDefenderConfig => {
+    if (!userConfig.defender || !userConfig.defender.apiKey || !userConfig.defender.apiSecret) {
+      const sampleConfig = JSON.stringify({ apiKey: 'YOUR_API_KEY', apiSecret: 'YOUR_API_SECRET' }, null, 2);
+      console.warn(
+        `Defender API key and secret are not set. Add the following to your hardhat.config.js exported configuration:\n${sampleConfig}\n`,
+      );
+    }
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return userConfig.defender!;
```

Comment:
Thoughts? It will still fail when validated by BaseApiClient.

```suggestion
    return userConfig.defender ?? { apiKey: '', apiSecret: '' };
```

Reply:
The type of `lazyObject` is correct. Because it uses a `Proxy` object under the hood it needs the lazy value to be object-like, which `undefined` isn't. I'm surprised you saw it work though, there is an error that is supposed to be [thrown](https://github.com/NomicFoundation/hardhat/blob/e0c2b2ae6bef0736eaf8f5d1c21eefe99777488d/packages/hardhat-core/src/internal/util/lazy.ts#L59-L61). `lazyObject(() => undefined)` fails when I try it.

Reply:
Good point, I tested with undefined keys, but with the `defender` object being set. I'll send a fix.

Reply:
Turns out removing extendConfig was the easiest way to go